### ISSUE TITLE
Infinite loop when script issues a query before the first layout in layout_2020

### ISSUE THREAD

**Issue by mrobinson** [servo/servo#25539]
Test case:
```
Text
<script>
    window.scrollTo(0, 0);
</script>
```

Working:
```
Text
<script>
    setTimeout(function() { window.scrollTo(0, 0); }, 5000);
</script>
```

Backtrace of the LayoutThread:
```
#0  0x000055555e4811e1 in core::mem::size_of () at /rustc/9b98af84c4aa66392236fff59c86da2130d46d46/src/libcore/mem/mod.rs:276
#1  0x000055555e4811e1 in core::ptr::swap_nonoverlapping_one (x=0x7ffe24bfad60, y=0x7ffe24bed5d0) at /rustc/9b98af84c4aa66392236fff59c86da2130d46d46/src/libcore/ptr/mod.rs:413
#2  0x000055555e481f5e in core::mem::swap (x=0x7ffe24bfad60, y=0x7ffe24bed5d0) at /rustc/9b98af84c4aa66392236fff59c86da2130d46d46/src/libcore/mem/mod.rs:556
#3  0x000055555e4822a1 in core::mem::replace (warning: (Internal error: pc 0x55555e47a690 in read in psymtab, but not in symtab.)
#4  0x000055555e47a691 in warning: (Internal error: pc 0x55555e47a690 in read in psymtab, but not in symtab.)
#5  0x000055555e47a61c in warning: (Internal error: pc 0x55555e47a61b in read in psymtab, but not in symtab.)
#6  0x000055555bc6e0b8 in <core::cell::BorrowRef as core::ops::drop::Drop>::drop (self=0x7ffe24bed6c8) at /rustc/9b98af84c4aa66392236fff59c86da2130d46d46/src/libcore/cell.rs:1141
#7  0x000055555ba0e40e in core::ptr::real_drop_in_place () at /rustc/9b98af84c4aa66392236fff59c86da2130d46d46/src/libcore/ptr/mod.rs:184
#8  0x000055555ba0fc17 in core::ptr::real_drop_in_place () at /rustc/9b98af84c4aa66392236fff59c86da2130d46d46/src/libcore/ptr/mod.rs:184
#9  0x000055555b8915dc in style::thread_state::get::{{closure}} () at components/style/thread_state.rs:76
#10 0x000055555bb8a5f1 in std::thread::local::LocalKey<T>::try_with (self=0x555560ef1098, f=...) at /rustc/9b98af84c4aa66392236fff59c86da2130d46d46/src/libstd/thread/local.rs:262
#11 0x000055555bb89e94 in std::thread::local::LocalKey<T>::with (self=0x555560ef1098, f=...) at /rustc/9b98af84c4aa66392236fff59c86da2130d46d46/src/libstd/thread/local.rs:239
#12 0x000055555bb02dde in style::thread_state::get () at components/style/thread_state.rs:71
#13 0x0000555558c20fcf in script::dom::bindings::root::LayoutDom<T>::downcast (self=0x7ffe24bed830) at components/script/dom/bindings/root.rs:370
#14 0x0000555556c8daf0 in <script::dom::bindings::root::LayoutDom<script::dom::node::Node> as script::dom::node::LayoutNodeHelpers>::composed_parent_node_ref (self=0x7ffe24bed8a0)
    at /home/martin/work/mozilla/servo/components/script/dom/node.rs:1262
#15 0x0000555556c3c4a3 in <layout_thread::dom_wrapper::ServoLayoutNode as style::dom::TNode>::parent_node (self=0x7ffe24bed8a0) at components/layout_thread_2020/dom_wrapper.rs:241
#16 0x0000555556c9e364 in <T as layout::dom_traversal::NodeExt>::parent_node (self=...) at /home/martin/work/mozilla/servo/components/layout_2020/dom_traversal.rs:369
#17 0x0000555556c9e7ca in <T as layout::dom_traversal::NodeExt>::unset_boxes_in_subtree (self=...) at /home/martin/work/mozilla/servo/components/layout_2020/dom_traversal.rs:436
#18 0x0000555556ddd315 in layout::dom_traversal::traverse_element (element=..., context=0x7ffe24bf0d08, handler=0x7ffe24bedda8) at /home/martin/work/mozilla/servo/components/layout_2020/dom_traversal.rs:99
#19 0x0000555556dddc00 in layout::dom_traversal::traverse_children_of (parent_element=..., context=0x7ffe24bf0d08, handler=0x7ffe24bedda8)
    at /home/martin/work/mozilla/servo/components/layout_2020/dom_traversal.rs:81
#20 0x0000555556ddd870 in layout::dom_traversal::NonReplacedContents::traverse (self=..., context=0x7ffe24bf0d08, node=..., inherited_style=0x7ffe24bee358, handler=0x7ffe24bedda8)
    at /home/martin/work/mozilla/servo/components/layout_2020/dom_traversal.rs:238
#21 0x0000555556e264b3 in layout::flow::construct::<impl layout::flow::BlockContainer>::construct
    (context=0x7ffe24bf0d08, root=..., block_container_style=0x7ffe24bee358, contents=..., content_sizes=layout::sizing::ContentSizesRequest::None)
    at /home/martin/work/mozilla/servo/components/layout_2020/flow/construct.rs:154
#22 0x0000555556dc64fe in layout::flow::construct::IntermediateBlockContainer::finish (self=..., context=0x7ffe24bf0d08, node=..., style=0x7ffe24bee358, content_sizes=layout::sizing::ContentSizesRequest::None)
    at /home/martin/work/mozilla/servo/components/layout_2020/flow/construct.rs:726
#23 0x0000555556dc563c in layout::flow::construct::BlockLevelJob<Node>::finish (self=..., context=0x7ffe24bf0d08, max_assign_in_flow_outer_content_sizes_to=...)
    at /home/martin/work/mozilla/servo/components/layout_2020/flow/construct.rs:642
#24 0x0000555556dc92d4 in layout::flow::construct::<impl layout::flow::BlockContainer>::construct::{{closure}} (acc=0x7ffe24bef4a0, creator=...)
    at /home/martin/work/mozilla/servo/components/layout_2020/flow/construct.rs:186
#25 0x0000555556dc91c8 in layout::flow::construct::<impl layout::flow::BlockContainer>::construct::{{closure}} (x=...) at /home/martin/work/mozilla/servo/components/layout_2020/flow/construct.rs:217
#26 0x0000555556c8a10c in core::iter::adapters::map_fold::{{closure}} (acc=(), elt=...) at /rustc/9b98af84c4aa66392236fff59c86da2130d46d46/src/libcore/iter/adapters/mod.rs:696
#27 0x0000555556dcfa1b in core::iter::traits::iterator::Iterator::fold::ok::{{closure}} (acc=(), x=...) at /rustc/9b98af84c4aa66392236fff59c86da2130d46d46/src/libcore/iter/traits/iterator.rs:1829
#28 0x0000555556c66b48 in core::iter::traits::iterator::Iterator::try_fold (self=0x7ffe24beeee0, init=(), f=...) at /rustc/9b98af84c4aa66392236fff59c86da2130d46d46/src/libcore/iter/traits/iterator.rs:1710
#29 0x0000555556c663dc in core::iter::traits::iterator::Iterator::fold (self=..., init=(), f=...) at /rustc/9b98af84c4aa66392236fff59c86da2130d46d46/src/libcore/iter/traits/iterator.rs:1832
#30 0x0000555556c8afc7 in <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (self=..., init=(), g=...)
    at /rustc/9b98af84c4aa66392236fff59c86da2130d46d46/src/libcore/iter/adapters/mod.rs:729
#31 0x0000555556c88d86 in core::iter::traits::iterator::Iterator::for_each (self=..., f=...) at /rustc/9b98af84c4aa66392236fff59c86da2130d46d46/src/libcore/iter/traits/iterator.rs:632
#32 0x0000555556c6f5e7 in <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::spec_extend (self=0x7ffe24bef188, iterator=...) at /rustc/9b98af84c4aa66392236fff59c86da2130d46d46/src/liballoc/vec.rs:2038
#33 0x0000555556c70e25 in <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (iterator=...) at /rustc/9b98af84c4aa66392236fff59c86da2130d46d46/src/liballoc/vec.rs:2021
#34 0x0000555556c73739 in <alloc::vec::Vec<T> as core::iter::traits::collect::FromIterator<T>>::from_iter (iter=...) at /rustc/9b98af84c4aa66392236fff59c86da2130d46d46/src/liballoc/vec.rs:1908
#35 0x0000555556c88a99 in core::iter::traits::iterator::Iterator::collect (self=...) at /rustc/9b98af84c4aa66392236fff59c86da2130d46d46/src/libcore/iter/traits/iterator.rs:1494
#36 0x0000555556e2687a in layout::flow::construct::<impl layout::flow::BlockContainer>::construct
    (context=0x7ffe24bf0d08, root=..., block_container_style=0x7ffe24bef840, contents=..., content_sizes=layout::sizing::ContentSizesRequest::None)
    at /home/martin/work/mozilla/servo/components/layout_2020/flow/construct.rs:214
#37 0x0000555556e26c4c in layout::flow::construct::<impl layout::flow::BlockFormattingContext>::construct
    (context=0x7ffe24bf0d08, node=..., style=0x7ffe24bef840, contents=..., content_sizes=layout::sizing::ContentSizesRequest::None) at /home/martin/work/mozilla/servo/components/layout_2020/flow/construct.rs:31
#38 0x0000555556de37ad in layout::formatting_contexts::IndependentFormattingContext::construct
    (context=0x7ffe24bf0d08, node=..., style=..., display_inside=layout::style_ext::DisplayInside::Flow, contents=..., content_sizes=layout::sizing::ContentSizesRequest::None)
37 0x0000555556e26c4c in layout::flow::construct::<impl layout::flow::BlockFormattingContext>::construct
    (context=0x7ffe24bf0d08, node=..., style=0x7ffe24bef840, contents=..., content_sizes=layout::sizing::ContentSizesRequest::None) at /home/martin/work/mozilla/servo/components/layout_2020/flow/construct.rs:31
#38 0x0000555556de37ad in layout::formatting_contexts::IndependentFormattingContext::construct
    (context=0x7ffe24bf0d08, node=..., style=..., display_inside=layout::style_ext::DisplayInside::Flow, contents=..., content_sizes=layout::sizing::ContentSizesRequest::None)
--Type <RET> for more, q to quit, c to continue without paging--  
    at /home/martin/work/mozilla/servo/components/layout_2020/formatting_contexts.rs:68
#39 0x0000555556e285e7 in layout::flow::root::construct_for_root_element (context=0x7ffe24bf0d08, root_element=...) at /home/martin/work/mozilla/servo/components/layout_2020/flow/root.rs:102
#40 0x0000555556e27f06 in layout::flow::root::BoxTreeRoot::construct (context=0x7ffe24bf0d08, root_element=...) at /home/martin/work/mozilla/servo/components/layout_2020/flow/root.rs:49
#41 0x0000555556db3058 in layout_thread::LayoutThread::handle_reflow::{{closure}} () at components/layout_thread_2020/lib.rs:1095
#42 0x0000555556c97eb7 in layout_thread::LayoutThread::handle_reflow (self=0x7ffe24bf3778, data=0x7ffe24bf17d8, possibly_locked_rw_data=0x7ffe24bf3738) at components/layout_thread_2020/lib.rs:1099
#43 0x0000555556db2f1a in layout_thread::LayoutThread::handle_request_helper::{{closure}} () at components/layout_thread_2020/lib.rs:706
```

**Comment by mrobinson** [comment]
cc @SimonSapin

**Comment by mrobinson** [comment]
I think this was fixed by #25560.
